src/express.ts

```diff
@@ -23,21 +26,64 @@ import { LruCache } from './utils/lru-cache';
 
 export { createExpressApp };
 
+function scenarioHasGroup(scenario: Scenario): scenario is Omit<
+	ScenarioWithOptionalProperties,
+	'group'
+> & {
+	group: string;
+} {
+	return !Array.isArray(scenario) && scenario.group != null;
+}
+
+function validateAllGroupsHaveNames({
+	scenarios,
+	groups,
+}: {
+	scenarios: ScenarioMap;
+	groups: Groups;
+}) {
+	const uniqueGroups = Array.from(
+		Object.values(scenarios)
+			.filter(scenarioHasGroup)
+			.reduce((set, { group }) => {
+				set.add(group);
+
+				return set;
+			}, new Set<string>())
+			.values(),
+	);
+
+	const groupsWithoutNames = uniqueGroups.filter((group) => !groups[group]);
+
+	if (groupsWithoutNames.length > 0) {
+		throw new Error(
+			`The following groups do not have a name: ${groupsWithoutNames.join(
+				', ',
+			)}`,
+		);
+	}
+}
+
 function createExpressApp({
 	scenarios: externalScenarioMap,
 	options = {},
+	groups = {},
 }: {
 	scenarios: ScenarioMap;
 	options?: Omit<Options, 'port'>;
+	groups?: Groups;
 }): ReturnType<typeof express> {
 	const {
 		uiPath = '/',
 		selectScenarioPath = '/select-scenario',
 		scenariosPath = '/scenarios',
+		groupsPath = '/groups',
 		cookieMode = false,
 		parallelContextSize = 10,
 	} = options;
 
+	validateAllGroupsHaveNames({ scenarios: externalScenarioMap, groups });
```

Comment:
Personally I'd not bother with this and just use the `id` for groups without names

Reply:
I did consider it, but ultimately added it for 2 reasons:

1. For those using the provided UI, it made sense to error early, because it gives fast feedback. 
2. It made the UI logic really easy. (although now I've written this I can see it's easy whatever I end up doing)

Reply:
I'll make it a console warning instead and do what you've suggested :+1: